Library/OcBootManagementLib/OcBootManagementLib.c

```diff
@@ -564,10 +564,10 @@ OcUnlockAppleFirmwareUI (
     //
     // Also acts as partial check that the memory is not something else entirely.
     //
-    if ((UINTN)(UIVars->mGopAlreadyConnected) != 1) {
-      DEBUG ((DEBUG_WARN, "OC: Cannot force reconnect Apple GOP %u\n", UIVars->mGopAlreadyConnected));
+    if (UIVars->mGopAlreadyConnected.FullFieldValue != 1) {
```

Comment:
@mhaeuser - Have you got time to confirm that this looks okay too? I thought this was what my previous check was doing, but re-reading it, realised it was not.

The change is not as clean a way to deal with what is actually a bool (padded to a QWORD), but I believe is a better check that we are in the memory we expect to be in.

Reply:
Thanks for offline discussion - modified and marking as closed.